### PR TITLE
Upstream arm64e and Pointer Authentication support

### PR DESCRIPTION
**Pull request by ahmedbougacha**
For a high-level overview, do read our RFC at http://lists.llvm.org/pipermail/llvm-dev/2019-October/136091.html

### CODE REVIEW

File: clang/include/clang/CodeGen/ConstantInitBuilder.h [hunk 1]
```diff
@@ -25,8 +25,11 @@
 #include <vector>
 
 namespace clang {
-namespace CodeGen {
+class GlobalDecl;
+class PointerAuthSchema;
```
Comment (im3ta) [inline clang/include/clang/CodeGen/ConstantInitBuilder.h]: ok

**Review by Whiskey911 — APPROVED** [review]

**Comment by ahmedbougacha** [comment]
@swift-ci test

**Comment by jpoumailloux** [comment]
Hi there, is there a place where progress upstreaming arm64e support can be tracked?

Upstreaming is a blocker to bringing arm64e support to Halide (https://github.com/halide/Halide/issues/4490).

**Comment by rjmccall** [comment]
We haven't set up a place to track that, no.  You could watch for commits from me or Ahmed on llvm-commits, but that's a high-volume list.

**Comment by Naville** [comment]
Hi:
Any progress update on this?
As far as I could tell LLVM upstream still only has bare Assembler support

**Comment by jpoumailloux** [comment]
Checking in if it would be possible to get an update and/or timeline on upstreaming of the remainder of arm64e support to LLVM. tyvm

**Pull request merged**